osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs

```diff
@@ -106,7 +109,10 @@ private void load(Storyboard storyboard)
 
         private void skinSourceChanged()
         {
-            Texture = skin.GetTexture(Sprite.Path) ?? textureStore.Get(Sprite.Path);
+            Texture = skin.GetTexture(Sprite.Path)
+                      // osu!classic is the default skin for storyboards. explicitly fall back to the osu!classic skin in case the skin provider does not provide osu!classic as a source (i.e. user does not have legacy skin selected).
+                      ?? skins.DefaultClassicSkin.GetTexture(Sprite.Path)
+                      ?? textureStore.Get(Sprite.Path);
```

Comment:
Compare: https://github.com/peppy/osu-stable-reference/blob/46cd3a10af7cc6cc96f4eba92ef1812dc8c3a27e/osu!/GameplayElements/Events/EventSprite.cs#L71-L74

- `TextureManager.Load(Filename)` fetches from beatmap skin first (unless user has beatmap skin disabled), followed by user skin, and finally osu!classic skin.
- `TextureManager.Load(Filename, SkinSource.Beatmap)` fetches explicitly from beatmap skin regardless of whether the user has beatmap skin enabled.

To match stable, osu!classic skin is placed directly after `skin.GetTexture`, so that the fetching order in lazer becomes:
  1. beatmap skin (handled by `skin.GetTexture`)
  2. user skin (handled by `skin.GetTexture`)
  3. default skin (handled by `skin.GetTexture` if skin is legacy, or `skins.DefaultClassicSkin.GetTexture` otherwise)
  4. storyboard resources (handled by `textureStore.Get`, is technically same as "beatmap skin")

This brings the behaviour of beatmap/storyboard resources being prioritised over user/default skin depending on whether the user has "beatmap skins" enabled. However, that isn't 100% the case because the `BeatmapSkinProvidingContainer` implementation shows that the beatmap skin will be ignored if `IsProvidingLegacyResources` is false. I'll leave that edge case for a future effort, if a storyboard gets affected by it.